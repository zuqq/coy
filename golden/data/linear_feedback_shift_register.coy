struct RandomF64(f64, i64);

fn next(u: i64) -> RandomF64 {
    // https://stackoverflow.com/a/17035915
    let v = (u >> 1) ^ (-(u & 1) & 2149580803);
    let x = ((v >> 8) as f64) / 16777216.0;
    RandomF64(x, v)
}

fn main_loop(u: i64, i: i64) -> () {
    if i <= 0 {
        ()
    } else {
        let RandomF64(x, v) = next(u);
        print!("{}\n", x);
        main_loop(v, i - 1)
    }
}

fn main() -> () {
    main_loop(1, 100)
}
