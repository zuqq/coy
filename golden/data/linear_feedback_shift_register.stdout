0.500488
0.750732
0.375366
0.688171
0.844574
0.422287
0.711632
0.856304
0.428152
0.714564
0.856794
0.428397
0.714687
0.856855
0.428427
0.714702
0.856863
0.428431
0.714704
0.856863
0.428432
0.714704
0.357352
0.178676
0.089338
0.544181
0.272090
0.636533
0.318267
0.159133
0.079567
0.539295
0.770136
0.885556
0.943266
0.471633
0.235816
0.617420
0.809198
0.905087
0.953032
0.476516
0.238258
0.618641
0.309320
0.154660
0.077330
0.038665
0.518844
0.259422
0.129711
0.565344
0.783160
0.391580
0.195790
0.598383
0.299192
0.149596
0.074798
0.037399
0.519188
0.259594
0.129797
0.565387
0.282693
0.640858
0.320429
0.660703
0.830839
0.915908
0.457954
0.228977
0.614977
0.307488
0.153744
0.576384
0.788680
0.893852
0.446926
0.722974
0.361487
0.681232
0.340616
0.170308
0.585642
0.292821
0.645922
0.822473
0.411236
0.205618
0.603297
0.301649
0.150824
0.075412
0.537218
0.268609
0.633816
0.816420
0.908698
0.454349
